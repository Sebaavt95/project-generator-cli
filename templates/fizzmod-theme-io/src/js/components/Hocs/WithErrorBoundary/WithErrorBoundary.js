import React from 'react';

const WithErrorBoundary = Component =>
	class ErrorBoundary extends React.PureComponent {
		constructor(props) {
			super(props);
			this.state = {
				hasError: false
			};
		}

		static getDerivedStateFromError() {
			return { hasError: true };
		}

		render() {
			const { hasError } = this.state;
			if (hasError) return null;
			return <Component {...this.props} />;
		}
	};

export default WithErrorBoundary;
