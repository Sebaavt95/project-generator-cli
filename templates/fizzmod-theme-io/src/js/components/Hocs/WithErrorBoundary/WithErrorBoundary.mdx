---
name: WithErrorBoundary
route: /components/Hocs/WithErrorBoundary
menu: Hocs
---

import { Playground, Props } from 'docz';
import Typography from './WithErrorBoundary';

# WithErrorBoundary

## Usage

```
import React, { useEffect } from 'react';
import WithErrorBoundary from 'components/Hocs/WithErrorBoundary';

const ComponentMakeApi = () => {
const [product, setProduct ] = useState({});

useEffect(() => {
	getProducts.then((res) => {
		setProduct(res)
	})
});
return (
	<div>
		<p>{product}</p>
	</div>
	)
}

export default WithErrorBoundary(ComponentMakeApi)

```