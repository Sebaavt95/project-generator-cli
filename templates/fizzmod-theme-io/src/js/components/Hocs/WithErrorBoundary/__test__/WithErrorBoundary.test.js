import React from 'react';
import { shallow } from 'enzyme';

import WithErrorBoundary from '../WithErrorBoundary';

describe('WithErrorBoundary without Error', () => {
	it('render ok', () => {
		const ChildComp = () => <div>test</div>;
		const Hoc = WithErrorBoundary(ChildComp);
		const wrapper = shallow(<Hoc />);
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find(ChildComp)).toHaveLength(1);
	});
});

describe('WithErrorBoundary with Error', () => {
	it('render null', () => {
		const childComp = () => <div>test</div>;
		const Hoc = WithErrorBoundary(childComp);
		const wrapper = shallow(<Hoc />);
		wrapper.setState({ hasError: true });
		expect(wrapper).toEqual({});
		expect(wrapper).toMatchSnapshot();
	});
});
