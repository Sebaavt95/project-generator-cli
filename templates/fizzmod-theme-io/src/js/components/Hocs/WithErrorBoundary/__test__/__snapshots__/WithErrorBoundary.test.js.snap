// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithErrorBoundary with Error render null 1`] = `""`;

exports[`WithErrorBoundary without Error render ok 1`] = `<ChildComp />`;
